edk2toollib/utility_functions.py

```diff
@@ -54,17 +54,21 @@ def Enum(*args):
 class PropagatingThread(threading.Thread):
     def run(self):
         self.exc = None
+        self.ret = None
         try:
             if hasattr(self, '_Thread__target'):
                 # Thread uses name mangling prior to Python 3.
                 self.ret = self._Thread__target(*self._Thread__args, **self._Thread__kwargs)
             else:
                 self.ret = self._target(*self._args, **self._kwargs)
+        except SystemExit as e:
+            self.ret = e.code
         except BaseException as e:
             self.exc = e
 
-    def join(self, timeout=None):
-        super(PropagatingThread, self).join()
+    def join(self, timeout=0.5):
```

Comment:
Is this timeout change to address its own issue?

Reply:
It's related- since we call join with a specific timeout in RunCmd, if we didn't pass it through it wouldn't work